zanata-war/src/main/java/org/zanata/service/impl/RequestServiceImpl.java

```diff
@@ -0,0 +1,172 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.zanata.service.impl;
+
+import com.google.common.collect.Lists;
+import org.jboss.seam.ScopeType;
+import org.jboss.seam.annotations.In;
+import org.jboss.seam.annotations.Name;
+import org.jboss.seam.annotations.Scope;
+import org.zanata.common.LocaleId;
+import org.zanata.dao.LanguageRequestDAO;
+import org.zanata.dao.RequestDAO;
+import org.zanata.events.RequestUpdatedEvent;
+import org.zanata.exception.RequestExistException;
+import org.zanata.model.HAccount;
+import org.zanata.model.HLocale;
+import org.zanata.model.LanguageRequest;
+import org.zanata.model.Request;
+import org.zanata.model.type.RequestState;
+import org.zanata.model.type.RequestType;
+import org.zanata.service.RequestService;
+import org.zanata.util.Event;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * @author Alex Eng <a href="aeng@redhat.com">aeng@redhat.com</a>
+ */
+@Name("requestServiceImpl")
+@Scope(ScopeType.STATELESS)
+public class RequestServiceImpl implements RequestService {
+
+    @In
+    private RequestDAO requestDAO;
+
+    @In
+    private LanguageRequestDAO languageRequestDAO;
+
+    @In("event")
+    private Event<RequestUpdatedEvent> requestUpdatedEvent;
+
+    @Override
+    public LanguageRequest createLanguageRequest(HAccount requester,
+            HLocale locale, boolean isRequestAsCoordinator,
+            boolean isRequestAsReviewer, boolean isRequestAsTranslator)
+            throws RequestExistException {
+        //search if there's any existing language request
+        //of the same requester, account and locale
+
+        LanguageRequest languageRequest =
+            languageRequestDAO.findRequesterPendingRequests(requester,
+                locale.getLocaleId());
+
+        if (languageRequest != null) {
+            throw new RequestExistException(
+                RequestType.LOCALE + ", " + requester.getUsername() + ", " +
+                    ", " + locale.getDisplayName());
+        }
+        String entityId = UUID.randomUUID().toString();
+        Request request = new Request(RequestType.LOCALE, requester, entityId);
+
+        languageRequest =
+            new LanguageRequest(request, locale, isRequestAsCoordinator,
+                isRequestAsReviewer, isRequestAsTranslator);
```

Comment:
It would be good to give LanguageRequest a builder, because I can see it being *extremely* easy to mix up the order of the booleans isRequestAsCoordinator, isRequestAsReviewer and isRequestAsTranslator.